src/services/AuthService.js

```diff
@@ -0,0 +1,137 @@
+import ENDPOINTS from "../endpoints";
+import { endpoints } from "../config";
+import httpService from "./HttpService";
+import jwt_decode from "jwt-decode";
+import { HTTP_METHODS, HTTP_STATUS_CODES } from '../constants';
+import { getItem,setItem,removeItem } from '../utils/localStorage';
+
+class AuthService {
+
+    httpService = httpService
+
+    constructor(){
+        this.init()
+    }
+
+    init = () => {
+        this.setAccessToken(this.getAccessToken())
+        this.httpService.addRequestInterceptor(this.checkTokenExpiration)
+        this.httpService.addResponseInterceptor(
+            this.handleSuccessResponse,
+            this.handleErrorResponse
+        )
+    }
+
+    register = async ({
+        firstName:name, 
+        email,
+        password,
+        confirmPassword: password_confirmation
+    }) => {
+
+        await this.httpService.request({
+            url: ENDPOINTS.REGISTER,
+            method: HTTP_METHODS.POST,
+            data: {name,email,password,password_confirmation}
+        })
+    }
+
+    setAccessToken = (token) => {
+        if(token){
+            setItem('access_token', token)
+
+            this.httpService.attachHeaders({
+                Authorization: `Bearer ${token}`
+            })
+        }
+    }
+
+    getAccessToken = () => getItem('access_token')
+
+    checkTokenExpiration = async (request) => {
+
+        if(request.url === endpoints.TOKEN_REFRESH){
+            return request
+        }
+
+        const token = this.getAccessToken()
+        
+        if(token && Date.now()/1000 >= jwt_decode(token).exp) {
+            const newToken = await this.refreshToken(token);
+            request.headers.Authorization = `Bearer ${newToken}`
+
+            return request
+        }
+
+        return request
+    }
+
+    refreshToken = async (old_token) => {
+        const {access_token: newToken } = await this.httpService.request({
+            url: endpoints.TOKEN_REFRESH,
+            method: HTTP_METHODS.POST,
+        })
+
+        this.setAccessToken(newToken)
+        return newToken
```

Comment:
pre returna uglavnom stavljamo jedan prazan red (osim ako to nije jedina linija u funkciji)